cookbook/views/api.py

```diff
@@ -661,6 +665,57 @@ def get_queryset(self):
         return queryset
 
 
+class AutoPlanViewSet(viewsets.ViewSet):
+    def create(self, request):
+        serializer = AutoMealPlanSerializer(data=request.data)
+
+        if serializer.is_valid():
+            keywords = serializer.validated_data['keywords']
+            start_date = serializer.validated_data['start_date']
+            end_date = serializer.validated_data['end_date']
+            meal_type = MealType.objects.get(pk=serializer.validated_data['meal_type_id'])
+            servings = serializer.validated_data['servings']
+
+            days = (end_date - start_date).days + 1
+            recipes = Recipe.objects.all()
+            meal_plans = list()
+
+            for keyword in keywords:
+                recipes = recipes.filter(keywords__name=keyword['name'])
+
+            recipes = recipes.order_by('?')[:days]
+            recipes = list(recipes)
+
+            for i in range(0, days):
+                day = start_date + datetime.timedelta(i)
+                recipe = random.choice(recipes)
```

Comment:
if i read this correctly this leaves the chance for a recipe to be choosen twice in the planning period, right?

I think I would just take the index of the day since the recipes are ordered randomly anyway so recipe = recipes[i]